logstash-core/lib/logstash/compiler/lscl.rb

```diff
@@ -0,0 +1,555 @@
+# encoding: utf-8
+require 'logstash/errors'
+require "treetop"
+require "logstash/compiler/treetop_monkeypatches"
+java_import Java::OrgLogstashConfigIr::DSL
+java_import Java::OrgLogstashConfigIr::SourceMetadata
+
+module LogStashCompilerLSCLGrammar; module LogStash; module Compiler; module LSCL; module AST
+  # Helpers for parsing LSCL files
+  module Helpers
+    def source_meta
+      line, column = line_and_column
+      Java::OrgLogstashConfigIr::SourceMetadata.new(source_file, line, column, self.text_value)
```

Comment:
Not just debug! This will be used to display the original files to users in all sorts of cases.